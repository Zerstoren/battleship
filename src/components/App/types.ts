import { ILobbyStore, IMainStore } from "../../stores/types";

export enum GameStatus {
  MAIN = 'main',
  LOBBY_LIST = 'list',
  CREATE_LOBBY = 'create',
  WAIT_CONNECT = 'wait',
  SET_SHIPS = 'set_ships',
  GAME = 'game_process',
  GAMEOVER = 'game_end',
}

export interface IGameState {
  gameStatus: GameStatus,
  gameWithUser: boolean,

  setGameStatus: (status: GameStatus) => void,
}

export interface IAppProps {
  mainStore?: IMainStore
}

export interface IProps {
  handleChangeGameStatus: (startGame: GameStatus) => void,
}

export interface ISetLobbyProps {
  handleSetLobby: (startGame: GameStatus, lobby: ILobbyStore) => void
}