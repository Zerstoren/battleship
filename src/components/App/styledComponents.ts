import styled from "styled-components";

export const CenteredDiv = styled.div`
  width: 600px;
  margin: 0 auto;
  text-align: center;
`;