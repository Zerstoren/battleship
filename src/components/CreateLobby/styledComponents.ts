import styled from "styled-components";

export const BlockSpacing = styled.div`
padding: 10px 0;
`;

export const FormFloating = styled.div.attrs(props => ({
  className: "form-floating"
}))``;

export const FloatingInput = styled.input.attrs(props => ({
  className: `form-control ${props.className}`
}))``;