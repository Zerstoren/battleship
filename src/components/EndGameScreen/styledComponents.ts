import styled from "styled-components";

export const ButtonComplete = styled.button.attrs((props) => ({
  className: 'btn btn-primary'
}))``;
