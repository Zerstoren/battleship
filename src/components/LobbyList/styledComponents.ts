import styled from "styled-components";

export const LobbyTable = styled.table.attrs(props => ({
  className: 'table table-striped table-hover'
}))``;