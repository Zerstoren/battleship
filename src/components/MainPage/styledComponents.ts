import styled from "styled-components";

export const ButtonSearchUser = styled.button.attrs(props => ({
  type: 'button',
  className: 'btn btn-secondary'
}))`
  margin: 0 12px;
`;
