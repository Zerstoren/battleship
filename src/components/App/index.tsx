import { inject, observer } from 'mobx-react';
import React, { FC } from 'react';
import { ILobbyStore } from '../../stores/lobby';
import CreateLobby from '../CreateLobby';
import LobbyList from '../LobbyList';
import Main from '../MainPage';
import SetShips from '../SetShips';
import WaitConnect from '../WaitConnect';
import { AlertError, CenteredDiv } from './styledComponents';
import { GameStatus, IAppProps } from './types';

const App: FC<IAppProps> = inject("mainStore")(observer((props) => {
  const state = props.mainStore;

  const handleGameStart = (startGame: GameStatus) => state!.setGameStatus(startGame);
  const handleSetLobby = (startGame: GameStatus, lobby: ILobbyStore) => state!.setLobby(startGame, lobby);
  const handleWaitLobby = (startGame: GameStatus, lobby: ILobbyStore | null = null) => state!.setLobby(startGame, lobby);
  const handleAcceptError = () => state?.setError('');

  let component = null;
  
  switch(state!.status) {
    case GameStatus.MAIN:
      component = (<Main handleChangeGameStatus={handleGameStart} />);
      break;
    
    case GameStatus.CREATE_LOBBY:
      component = (<CreateLobby handleChangeGameStatus={handleWaitLobby} />);
      break;

    case GameStatus.WAIT_CONNECT:
      component = (<WaitConnect handleChangeGameStatus={handleGameStart} />);
      break;

    case GameStatus.LOBBY_LIST:
      component = (<LobbyList handleSetLobby={handleSetLobby} />);
      break;

    case GameStatus.SET_SHIPS:
      component = (<SetShips handleChangeGameStatus={handleGameStart} />);
  }

  let err = null;
  
  if (state!.error) {
    err = <AlertError>
      <strong>{state!.error}</strong>
      <button type="button" className="btn-close" onClick={handleAcceptError}></button>
    </AlertError>;
  }

  return (
    <>
      {err}
      <CenteredDiv>
        {component}
      </CenteredDiv>
    </>
  );
}));

export default App;
