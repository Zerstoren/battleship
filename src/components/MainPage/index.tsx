import React, { FC } from "react";
import { AppHeader } from "../../shared/StyledComponents/Headers";
import { GameStatus, IProps } from "../App/types";
import { ButtonSearchUser } from "./styledComponents";

const Main: FC<IProps> = ({handleChangeGameStatus: handleStartGame}) => {
  return (
    <>
      <AppHeader>Battleship</AppHeader>
      <div>
        <ButtonSearchUser onClick={() => handleStartGame(GameStatus.LOBBY_LIST)}>
          Search opponent
        </ButtonSearchUser>

        <ButtonSearchUser onClick={() => handleStartGame(GameStatus.CREATE_LOBBY)}>
          Create lobby
        </ButtonSearchUser>
      </div>
    </>
  );
}

export default Main;