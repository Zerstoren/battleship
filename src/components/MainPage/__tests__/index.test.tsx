import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";
import MainPage from '..';
import { ButtonCreateLobby, ButtonSearchUser } from '../styledComponents';
import { GameStatus } from '../../App/types';

it('render main page', () => {
  const tree = renderer
    .create(<MainPage handleChangeGameStatus={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('check create lobby', () => {
  let gGameStatus = null;
  const handleGameStatus = (gameStatus: GameStatus) => gGameStatus = gameStatus;


  const wrapper = shallow(<MainPage handleChangeGameStatus={handleGameStatus} />);
  wrapper.find(ButtonCreateLobby).simulate('click');
  expect(gGameStatus).toEqual(GameStatus.CREATE_LOBBY);
});

it('check search lobby', () => {
  let gGameStatus = null;
  const handleGameStatus = (gameStatus: GameStatus) => gGameStatus = gameStatus;

  const wrapper = shallow(<MainPage handleChangeGameStatus={handleGameStatus} />);
  wrapper.find(ButtonSearchUser).simulate('click');
  expect(gGameStatus).toEqual(GameStatus.LOBBY_LIST);
});
