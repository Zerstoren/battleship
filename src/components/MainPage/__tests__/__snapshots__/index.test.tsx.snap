// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render main page 1`] = `
Array [
  <h1
    className="sc-bdfBwQ display-3"
  >
    Battleship
  </h1>,
  <div>
    <button
      className="sc-gsTCUz fGVasU btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Search opponent
    </button>
    <button
      className="sc-gsTCUz sc-dlfnbm fGVasU btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Create lobby
    </button>
  </div>,
]
`;
